js/apps/system/_admin/aardvark/APP/frontend/js/views/loginView.js

```diff
@@ -135,61 +119,59 @@
       event.preventDefault();
       this.clear();
 
-      var username = $('#loginUsername').val();
-      var password = $('#loginPassword').val();
+      const username = $('#loginUsername').val();
+      const password = $('#loginPassword').val();
 
       if (!username) {
         // do not send unneccessary requests if no user is given
         return;
       }
 
-      this.collection.login(username, password, this.loginCallback.bind(this, username, password));
+      this.collection.login(username, password, () => this.loginCallback(username, password));
     },
 
     loginCallback: function (username, password, error) {
-      var self = this;
-
       if (error) {
         if (self.loginCounter === 0) {
-          self.loginCounter++;
-          self.collection.login(username, password, this.loginCallback.bind(this, username));
+          this.loginCounter++;
+          this.collection.login(username, password, () => this.loginCallback(username));
           return;
         }
-        self.loginCounter = 0;
+        this.loginCounter = 0;
         $('.wrong-credentials').show();
         $('#loginDatabase').html('');
         $('#loginDatabase').append(
           '<option>_system</option>'
         );
       } else {
-        self.renderDBSelection(username);
+        this.renderDBSelection(username);
       }
     },
 
     renderDBSelection: function (username) {
-      var self = this;
-      var url = arangoHelper.databaseUrl('/_api/user/' + encodeURIComponent(username) + '/database', '_system');
-
-      if (frontendConfig.authenticationEnabled === false) {
-        url = arangoHelper.databaseUrl('/_api/database/user');
-      }
-
       $('.wrong-credentials').hide();
-      self.loggedIn = true;
+      this.loggedIn = true;
 
       // get list of allowed dbs
       $.ajax({
         type: "GET",
-        url: url,
-        success: function (permissions) {
+        url: frontendConfig.authenticationEnabled
+          ? arangoHelper.databaseUrl(`/_api/user/${encodeURIComponent(username)}/database`, '_system')
+          : arangoHelper.databaseUrl('/_api/database/user'),
```

Comment:
That url build logic could be moved into a helper method as also used here:
- https://github.com/arangodb/arangodb/pull/18226/files#diff-cf85c1e65271ee5d2a8659056f3125557933c993640a3a4a7afee2d96a091ac6R71